### PR TITLE
libreoffice: add module

### LINKED ISSUE #5612: Possible LibreOffice `programs` option?
### Description
LibreOffice, the de-facto productivity software suite, already has a [package available in `nixpkgs`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/office/libreoffice), but the current status quo doesn't allow for declarative customisation of any of the facets of LibreOffice (e.g. extensions, preferences) (see: https://github.com/NixOS/nixpkgs/issues/235416) 

Having a `home-manager` option would make for a simplistic declarative configuration of such configurations.

Caveats however, are that currently no LibreOffice extensions are packaged for Nix; however, with the exception of the Zotero reference manager, where the extension is currently bundled with the full package on `nixpkgs`.

### PR DESCRIPTION
**Pull request by ekhh**
### Description

Closes #5612.

Adds `programs.libreoffice` for [LibreOffice](https://www.libreoffice.org/), a comprehensive office productivity suite. Currently, extensions are not covered because none have been packaged; the `.xcu` file should cover all settings, however.

This is my first time writing a module. Feedbacks are appreciated.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `nix fmt` or
      `nix-shell -A dev --run treefmt`.

- [x] Code tested through `nix build .#test-all`
      or a targeted `nix run .#tests -- <pattern>`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

  ```
  {component}: {description}

  {long description}
  ```

  See [CONTRIBUTING](https://nix-community.github.io/home-manager/#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/a51598236f23c89e59ee77eb8e0614358b0e896c/modules/programs/lesspipe.nix#L11).
- [x] Generate a news entry. See [News](https://nix-community.github.io/home-manager/index.xhtml#sec-news)
- [x] Basic tests added. See [Tests](https://nix-community.github.io/home-manager/index.xhtml#sec-tests)

### CODE REVIEW

**Comment by ekhh** [comment]
Does it not recognize my username?

**Comment by ekhh** [comment]
Upstream [replaced libfreoffice-fresh and libreoffice-still](https://github.com/NixOS/nixpkgs/commit/38552f4113ffe567d2a16d8d2a6407d6b42a7ee7).

File: modules/programs/libreoffice.nix [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+  options.programs.libreoffice = {
+    enable = lib.mkEnableOption "LibreOffice, a comprehensive office suite.";
+    package = lib.mkPackageOption pkgs "libreoffice" {
+      nullable = true;
+      example = "pkgs.libreoffice-qt";
+    };
+    settings = lib.mkOption {
+      default = { };
+      type = lib.types.attrsOf lib.types.anything;
```
Comment (khaneliman) [inline modules/programs/libreoffice.nix]: issue (blocking): make the option type match the XCU serializer

The renderer requires two attribute levels (`path -> property -> value`) and only serializes scalar values. Encode that contract in the option type instead of accepting `anything`:

```nix
let
  settingValueType = with lib.types; oneOf [ bool int float str ];
in
lib.mkOption {
  type = lib.types.attrsOf (lib.types.attrsOf settingValueType);
  default = { };
  # ...
}
```

This is the same primitive set used for free-form settings in [`bspwm/options.nix`](https://github.com/nix-community/home-manager/blob/6840c9a8f50722944eb106ce8bb237c138584f2a/modules/services/window-managers/bspwm/options.nix#L4-L12). Add evaluation coverage rejecting both a scalar directly under a path and a list-valued property, plus generation coverage for Boolean, integer, float, and string leaves.

File: modules/programs/libreoffice.nix [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      + ''xmlns:xs="http://www.w3.org/2001/XMLSchema" ''
+      + ''xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">''
+      + "\n"
+      +
+        lib.concatMapAttrsStringSep ""
+          (
+            path: attrs:
+            lib.concatMapAttrsStringSep "" (
+              name: value:
+              lib.optionalString (toValue value != "") (
```
Comment (khaneliman) [inline modules/programs/libreoffice.nix]: issue (blocking): serialize explicit empty strings

`lib.optionalString (toValue value != "")` drops a valid configured value. LibreOffice itself declares [`FilePickerLastService` as a string](https://github.com/LibreOffice/core/blob/785947d2f93ea4a64a9be5bd95450e6698a3cd6b/officecfg/registry/schema/org/openoffice/Office/Common.xcs#L5291-L5294) and supplies [an empty `<value/>` default](https://github.com/LibreOffice/core/blob/785947d2f93ea4a64a9be5bd95450e6698a3cd6b/officecfg/registry/data/org/openoffice/Office/Common.xcu#L418-L420). Remove the `optionalString` guard and emit the `<item>` for every value admitted by `settingValueType`; `value = ""` will then produce `<value></value>`.

Add `FilePickerLastService = ""` to the generated-file test and assert that its property remains present with an empty value.

File: modules/programs/libreoffice.nix [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+          )
+          (
+            lib.recursiveUpdate (lib.optionalAttrs (cfg.package != null) {
+              ${ooSetupPath}.${ooSetupKey} =
+                /*
+                  This value records the last major version (xx.xx) whereupon a
+                  setup was run. The configuration will be ignored if it is
+                  absent.
+                */
+                lib.head (lib.match "^([^.]+[.][^.]+).*" cfg.package.version);
```
Comment (khaneliman) [inline modules/programs/libreoffice.nix]: issue (blocking): validate package-version derivation before generating the setup value

Do not read `cfg.package.version` or call `lib.head` until both the attribute and match are known to exist. Compute a nullable derived value, then require the existing explicit setting when derivation fails:

```nix
packageVersion = if cfg.package == null then null else cfg.package.version or null;
versionMatch =
  if packageVersion == null then null
  else builtins.match "^([0-9]+[.][0-9]+)([.].*)?$" packageVersion;
derivedSetupVersion =
  if versionMatch == null then null else builtins.head versionMatch;
hasExplicitSetupVersion =
  lib.hasAttrByPath [ ooSetupPath ooSetupKey ] cfg.settings;
```

Assert that empty settings, an explicit setup version, or a non-null `derivedSetupVersion` exists. Only inject `${ooSetupPath}.${ooSetupKey} = derivedSetupVersion` when it is non-null, while preserving the current explicit-setting precedence. Add tests for a package without `version`, a malformed version, and an explicit `ooSetupLastVersion` that permits each package to evaluate.

**Review by khaneliman — CHANGES_REQUESTED** [review]
Please split the new maintainer entry from the module addition commits.

File: modules/programs/libreoffice.nix (hunk 1 above)
Comment (ekhh) [reply modules/programs/libreoffice.nix]: Updated `type`.

File: modules/programs/libreoffice.nix (hunk 2 above)
Comment (ekhh) [reply modules/programs/libreoffice.nix]: Removed `lib.optionalString` and updated `nmt.script`.

File: modules/programs/libreoffice.nix (hunk 3 above)
Comment (ekhh) [reply modules/programs/libreoffice.nix]: Added the aforementioned checks.

File: modules/programs/libreoffice.nix [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      (lib.hm.assertions.assertPlatform "programs.libreoffice" pkgs lib.platforms.linux)
+      {
+        assertion = cfg.settings == { } || ooSetup.hasExplicitVersion || ooSetup.generatedVersion != null;
+        message = "`\"${ooSetup.versionPath}\".${ooSetup.versionKey}` must be set in `programs.libreoffice.settings`.";
+      }
+    ];
+    home.packages = lib.mkIf (cfg.package != null) [ cfg.package ];
+    # `builtins.toXML` has various issues and may be deprecated in the future;
+    # see <https://github.com/NixOS/nix/issues/15745#issue-4334624253>.
+    xdg.configFile.libreoffice-configuration = lib.mkIf (cfg.settings != { }) {
```
Comment (khaneliman) [inline modules/programs/libreoffice.nix:84]: LibreOffice replaces this symlink with a regular file when it writes `registrymodifications.xcu`. I reproduced that with LibreOffice 26.2.5.2 in an isolated profile. The rewritten file had new LibreOffice entries, so the next Home Manager activation fails in `checkLinkTargets`.

```suggestion
    xdg.configFile.libreoffice-configuration = lib.mkIf (cfg.settings != { }) {
      force = true;
```

This matches [HyprPanel](https://github.com/nix-community/home-manager/blob/7dc7b2dc9c2e2fce0f9ed4c7f49b124841aa34bb/modules/programs/hyprpanel/default.nix#L90-L95), which sets `force = true` because the application rewrites its generated config. The [Firefox search option](https://github.com/nix-community/home-manager/blob/7dc7b2dc9c2e2fce0f9ed4c7f49b124841aa34bb/modules/programs/firefox/profiles/search.nix#L439-L448) documents the same symlink-replacement behavior and warns that forcing discards runtime changes.

Please document that tradeoff here. Add a regression test that puts a different regular file at this path and verifies that activation restores the managed link.

**Review by khaneliman — CHANGES_REQUESTED** [review]

File: modules/programs/libreoffice.nix (hunk 4 above)
Comment (ekhh) [reply modules/programs/libreoffice.nix:84]: Added `force`, placed a warning in the option description, updated the NMT test case, and documented its behaviour.

**Comment by nyabinary** [comment]
This probably should have provided an ``extraPackages`` for extension requiring a binary for example and a languages setting to choose (and download) the associated hunspell package for the dictionary
